plasmapy/analysis/signal.py

```diff
@@ -0,0 +1,45 @@
+"""_summary_
+
+"""
+
+import numpy as np
+import scipy.io as spio
+import scipy.integrate as sp
+from scipy import signal
+from warnings import warn
+
+
+def remove_offset(data_array: np.ndarray, start_idx: int, end_idx: int) ->  np.ndarray:
+   """_summary_ : Function which preforms mean subtraction
+   
+   Args:     
+      data_array (array): array containing values from which to mean sub
+      start_idx (int): idx of where to start obtaining mean from
+      end_idx (int): idx of where to stop obtaining mean from
+   Returns:
+      data_array (array): array after mean_sub is preformed 
+   """
+   data_array = np.array(data_array)
+   meansub = np.mean(data_array[start_idx:end_idx])
+   data_array[:] = [x - meansub for x in data_array]
+   return data_array
+
+def band_pass_filter(data_array: np.ndarray, cutoff: float, sampling_frequency: float, order: int) ->  np.ndarray:
```

Comment:
This function implements a high-pass filter, not a band-pass filter. Consider renaming this function "high_pass_filter" or implement the high and low cutoffs